Issue when endpoint is not reachable

Using provider 0.7.0 there still seems to be some issues if the endpoint is not reachable

```sh
Stack trace from the terraform-provider-sonatyperepo_v0.7.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xa392d7]

goroutine 12 [running]:
terraform-provider-sonatyperepo/internal/provider/common.getResponseBody(0x0)
	terraform-provider-sonatyperepo/internal/provider/common/api.go:79 +0x17
terraform-provider-sonatyperepo/internal/provider/common.HandleApiError({0x1090dd1, 0x38}, 0xc0000d91d8, 0x0, 0xc0000bc970)
	terraform-provider-sonatyperepo/internal/provider/common/api.go:41 +0x16b
terraform-provider-sonatyperepo/internal/provider/common.(*SonatypeDataSourceData).CheckWritableAndGetVersion(0xc0000d9300, {0x11d5388, 0xc0000adcb0}, 0xc0000bc970)
	terraform-provider-sonatyperepo/internal/provider/common/model.go:43 +0x365
terraform-provider-sonatyperepo/internal/provider.(*SonatypeRepoProvider).Configure(0xc0001cd860, {0x11d5388, 0xc0000adcb0}, {{0xc0000127b8, 0x6}, {{{0x11da608, 0xc0003965a0}, {0xeed8c0, 0xc000396450}}, {0x11dd988, ...}}, ...}, ...)
	terraform-provider-sonatyperepo/internal/provider/provider.go:183 +0x9f5
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0xc0002f8008, {0x11d5388, 0xc0000adcb0}, 0xc0000bc9b0, 0xc0000bc960)
	github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_configureprovider.go:18 +0x132
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ConfigureProvider(0xc0002f8008, {0x11d5388?, 0xc0000adbc0?}, 0xc0000904e0)
	github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/proto6server/server_configureprovider.go:39 +0x2a5
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ConfigureProvider(0xc000212e60, {0x11d5388?, 0xc0000acff0?}, 0xc0000bc640)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:559 +0x342
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ConfigureProvider_Handler({0x10280e0, 0xc000212e60}, {0x11d5388, 0xc0000acff0}, 0xc0000ca200, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:557 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00020e200, {0x11d5388, 0xc0000acf60}, 0xc0000e0ba0, 0xc0002f27b0, 0x19837f0, 0x0)
	google.golang.org/grpc@v1.73.0/server.go:1405 +0x103b
google.golang.org/grpc.(*Server).handleStream(0xc00020e200, {0x11d5d38, 0xc0000f2000}, 0xc0000e0ba0)
	google.golang.org/grpc@v1.73.0/server.go:1815 +0xbaa
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.73.0/server.go:1035 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 8
	google.golang.org/grpc@v1.73.0/server.go:1046 +0x125

Error: The terraform-provider-sonatyperepo_v0.7.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,8 @@
 
 ## X.Y.Z (UNRELEASED)
 
+BUG FIXES:
+* **Bug:** Further improve connectivity errors to Sonatype Nexus Repository without the provider crashinh [GH-105]
 
 ## 0.8.0 October 18, 2025
 

--- a/internal/provider/common/api.go
+++ b/internal/provider/common/api.go
@@ -23,6 +23,7 @@ import (
 	"log"
 	"net"
 	"net/http"
+	"reflect"
 	"syscall"
 
 	"github.com/hashicorp/terraform-plugin-framework/diag"
@@ -36,10 +37,18 @@ func HandleApiError(message string, err *error, httpResponse *http.Response, res
 			fmt.Sprintf("Networking Error: %s (%v)", errorMessage, *err),
 		)
 	} else {
-		respDiags.AddError(
-			message,
-			fmt.Sprintf("%s: %s: %s", *err, httpResponse.Status, getResponseBody(httpResponse)),
-		)
+		if httpResponse != nil {
+			respDiags.AddError(
+				message,
+				fmt.Sprintf("%s: %s: %s", *err, httpResponse.Status, getResponseBody(httpResponse)),
+			)
+		} else {
+			respDiags.AddError(
+				message,
+				fmt.Sprintf("Unexpected Error: %v ('%s'): ", *err, reflect.TypeOf(*err)),
+			)
+
+		}
 	}
 }
 

